app/controllers/v1/profiles_controller.rb

```diff
@@ -27,13 +34,24 @@ def create
       profile = Profile.new(profile_create_attributes).fill_from_parent
 
       if profile.save
+        profile.update_hosts(new_host_ids)
         profile.update_rules(ids: new_rule_ids)
         render_json profile, status: :created
       else
         render_error profile
       end
     end
 
+    def update
+      if profile.update(profile_update_attributes)
+        profile.update_hosts(new_host_ids)
+        profile.update_rules(ids: new_rule_ids)
+        render_json profile
+      else
+        render_error profile
```

Comment:
I believe this should be rescuing any `StandardError` in here - e.g, if `profile.update` succeeds but either of `profile.update_hosts` or `profile.update_rules` fail, it's a 500 instead of a captured error that we can return to the user.

Reply:
True, but how exactly might `update_hosts` or `update_rules` fail? If we're catching exceptions, we have to figure out which specific exceptions are valid and rescue those errors. For example, `import!` might raise a `ActiveRecord::RecordNotUnique`, though we've implemented it in a way to always avoid that scenario. I've added a global `rescue_from` in the application_controller in this PR to address that: https://github.com/RedHatInsights/compliance-backend/pull/510/files#r451533427 Any other errors?